### DISCUSSION TITLE
Building WinUIGallery.DesktopWap gives NETSDK1005 error

### DISCUSSION THREAD

**Discussion by rcohn** [microsoft/WinUI-Gallery#1147] [Q&A] [answered]
Hello -

I downloaded the WinUIGallery sources and attempted to build WinUIGallery.DesktopWap. I'm getting this error:

`1>------ Build started: Project: WinUIGallery.DesktopWap, Configuration: Debug x64 ------
Failed to restore C:\Users\rcohn\Downloads\WinUI-Gallery-main\WinUI-Gallery-main\WinUIGallery\WinUIGallery.DesktopWap.csproj (in 11 ms).
Failed to restore C:\Users\rcohn\Downloads\WinUI-Gallery-main\WinUI-Gallery-main\WinUIGallery\WinUIGallery.DesktopWap.Package.wapproj (in 58 ms).
NuGet package restore failed. Please see Error List window for detailed warnings and errors.
1>C:\Program Files\dotnet\sdk\6.0.307\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(267,5): error NETSDK1005: Assets file 'C:\Users\rcohn\Downloads\WinUI-Gallery-main\WinUI-Gallery-main\WinUIGallery\obj\WinUIGallery.DesktopWap\project.assets.json' doesn't have a target for 'net6.0'. Ensure that restore has run and that you have included 'net6.0' in the TargetFrameworks for your project.
1>Done building project "WinUIGallery.DesktopWap.csproj" -- FAILED.`

As near as I understand it, this is some kind of NuGet package error.  I attempted to do a "Restore NuGet packages", but that failed.

I'm using Visual Studio 2022 17.2.5 on Windows 11.  Could you please give some advice here?  

Thanks very much.
Robert

**Comment by marcelwgn** [comment, marked as answer]
Solution found by @rcohn:

However, since my last comment, it seems I have found a solution. There were some suggestions in: https://stackoverflow.com/questions/56558843/local-source-of-nuget-packages-doesnt-exist, one of which helped:
dotnet new nugetconfig
followed by:
dotnet restore WinUIGallery.DesktopWap.csproj

So now I am able to build WinUIGallery.DesktopWap and publish it. The dotnet restore, by itself, did not work until I did the dotnet new command. For some reason project.assets.json had an empty "targets" field before the nuget commands. Now, "targets" is filled in and targeted to net6.0-windows10.0.18362.

------


Did you install the [Windows App SDK extension](https://aka.ms/windowsappsdk/1.2/latest/WindowsAppSDK.Cs.Extension.Dev17.Standalone.vsix)? I also ran into an issue restoring the packages and installing the extension fixed this issue for me.

**Reply by rcohn** [reply]
Installing WindowsAppSDK.Cs.Extension.Dev17.Standalone.vsix does not help.  I am still getting the error.
Thank you for the reply anyway.

**Reply by marcelwgn** [reply]
Hmm I see. Are you able to create a new WinUI 3 app and build that or is it just the WinUI Gallery that is not restoring correctly?

**Reply by rcohn** [reply]
Hello again -

We have been building a production-level WinUI 3 desktop app for the last year.  And I have created a number of demo WinUI 3 apps.  The problem was with building the WinUI Gallery, which I wanted to build so that I can get some understanding of other WinUI element usage.

However, since my last comment, it seems I have found a solution.  There were some suggestions in: https://stackoverflow.com/questions/56558843/local-source-of-nuget-packages-doesnt-exist, one of which helped:
`dotnet new nugetconfig`
followed by:
`dotnet restore WinUIGallery.DesktopWap.csproj`

So now I am able to build WinUIGallery.DesktopWap and publish it.  The dotnet restore, by itself, did not work until I did the dotnet new command.  For some reason project.assets.json had an empty "targets" field before the nuget commands.  Now, "targets" is filled in and targeted to net6.0-windows10.0.18362.

Since I have seen a number of submissions of problems the same as mine or very similar, it might be a good idea to see if there is any way to ensure that the solution no longer has this problem - or at least provide some guidance in the Readme on how to resolve it.

Again, I appreciate your follow-up. 

Best regards.

**Reply by marcelwgn** [reply]
Great to hear that you were able to get the project compiling! Would you like to post your comment as top level answer so it can be marked as solution?

**Reply by rcohn** [reply]
Yes, that would be fine.  Is there a way I can do that, or do you do that as moderator?
Thanks.

**Reply by marcelwgn** [reply]
Since you asked the question, you can also mark your answer as the solution. Otherwise, I would also be more than happy to do that for you if it doesn't work.

**Reply by rcohn** [reply]
I'm not seeing a way to mark my reply as the answer.  I see only the one "mark as answer" button in one of your previous comments.  There doesn't seem to be a way to mark a comment within a thread as the answer.  But anyway, I pressed the button and people can read further to get the actual answer.

**Reply by marcelwgn** [reply]
Unfortunately, you cannot mark a comment to an answer as solution, that's why I suggested posting your comment as a top level answer. Anyways, I've updated my comment to show the solution you found.
